### PR TITLE
Swap endpoints for flow

### PR DESCRIPTION
**Pull request by Levitan**
New API for swap enpoints for flows

### CODE REVIEW

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowController.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     * The flow service instance.
+     */
+    @Autowired
+    private FlowService flowService;
+
+    /**
+     * Bulk update for flow.
+     */
+    @ApiOperation(value = "Swap flow endpoints", response = SwapFlowEndpointPayload.class)
+    @PatchMapping("/swap-endpoint")
```
Comment (sergii-iakovenko) [inline services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowController.java]: As per api-guidelines.md, POST method should be used for command pattern.

File: services/src/messaging/src/main/java/org/openkilda/messaging/payload/flow/SwapFlowEndpointPayload.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+@Data
+@EqualsAndHashCode
+@AllArgsConstructor
+@NoArgsConstructor
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(SnakeCaseStrategy.class)
+public class SwapFlowEndpointPayload {
```
[the remark this answers is not in the document]
Comment (Levitan) [reply services/src/messaging/src/main/java/org/openkilda/messaging/payload/flow/SwapFlowEndpointPayload.java]: Moved

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowController.java (hunk 1 above)
Comment (Levitan) [reply services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowController.java]: Changed to POST method

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowController.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+public class FlowController extends BaseController {
+
+    /**
+     * The logger.
+     */
+    private static final Logger logger = LoggerFactory.getLogger(FlowController.class);
+
+    /**
+     * The flow service instance.
+     */
```
Comment (sergii-iakovenko) [inline services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v2/FlowController.java]: Same as above.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        commandFactory = new FlowCommandFactory();
+        repositoryFactory = persistenceManager.getRepositoryFactory();
+        AvailableNetworkFactory availableNetworkFactory =
+                new AvailableNetworkFactory(pathComputerConfig, repositoryFactory);
+        pathComputerFactory = new PathComputerFactory(pathComputerConfig, availableNetworkFactory);
+        flowValidator = new FlowValidator(repositoryFactory);
+        flowResourcesManager = new FlowResourcesManager(persistenceManager, flowResourcesConfig);
+        flowService = new FlowService(persistenceManager, pathComputerFactory, flowResourcesManager,
+                flowValidator, commandFactory);
+        featureTogglesService = new FeatureTogglesService(persistenceManager.getRepositoryFactory());
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java]: repositoryFactory (obtained before) can be used here.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+import org.apache.storm.topology.base.BaseRichBolt;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class FlowOperationsBolt extends BaseRichBolt {
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java]: Shoudn't it be implemented following hub&spoke approach?

Comment (sergii-iakovenko) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java]: Could you create a task to re-implement swap operation in flowhs topology?

Comment (Levitan) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java]: New issue created https://github.com/telstra/open-kilda/issues/2449

File: services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+            featureTogglesService.checkFeatureToggleEnabled(FeatureToggle.UPDATE_FLOW);
+
+            SwapFlowEndpointRequest request = (SwapFlowEndpointRequest) message.getData();
+
+            List<FlowPair> flowPairs =
+                    flowService.swapFlowEnpoints(FlowMapper.INSTANCE.buildFlow(request.getFirstFlow()),
+                            FlowMapper.INSTANCE.buildFlow(request.getSecondFlow()),
+                            new FlowCommandSenderImpl(message.getCorrelationId(), tuple, StreamType.UPDATE));
+
+            logger.info("Swap endpoint");
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/FlowOperationsBolt.java]: Could we log some details from the first and second flows?

**Pull request merged**